Missing typing_extensions dependency

Hey,

In the last release (https://github.com/mdomke/python-ulid/compare/3.0.0...3.1.0) you added a import from typing_extensions but forgot to update the dependencies, so now on a fresh install, unless I specify it on my requirements, I'm getting the following error:

```
 from ulid import ULID
 /usr/local/lib/python3.13/site-packages/ulid/__init__.py:16: in <module>
 from typing_extensions import Self
 E   ModuleNotFoundError: No module named 'typing_extensions' 
```

Cheers!

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -26,6 +26,9 @@ classifiers = [
     "Programming Language :: Python :: 3.13",
     "Topic :: Software Development :: Libraries",
 ]
+dependencies = [
+    "typing-extensions; python_version < '3.11'",
+]
 
 [project.urls]
 Homepage = "https://github.com/mdomke/python-ulid"

--- a/ulid/__init__.py
+++ b/ulid/__init__.py
@@ -2,7 +2,6 @@ from __future__ import annotations
 
 import functools
 import os
-import sys
 import time
 import uuid
 from datetime import datetime
@@ -14,22 +13,22 @@ from typing import Generic
 from typing import TYPE_CHECKING
 from typing import TypeVar
 
-
-if sys.version_info >= (3, 11):
-    from typing import Self
-else:
-    from typing_extensions import Self
-
 from ulid import base32
 from ulid import constants
 
 
 if TYPE_CHECKING:  # pragma: no cover
+    import sys
     from collections.abc import Callable
 
     from pydantic import GetCoreSchemaHandler
     from pydantic import ValidatorFunctionWrapHandler
     from pydantic_core import CoreSchema
+
+    if sys.version_info >= (3, 11):
+        from typing import Self
+    else:
+        from typing_extensions import Self
 
 try:
     from importlib.metadata import version
@@ -204,7 +203,7 @@ class ULID:
             a value when they're unsure what format/primitive type it will be given in.
         """
         if isinstance(value, ULID):
-            return cast(Self, value)
+            return cast("Self", value)
         if isinstance(value, uuid.UUID):
             return cls.from_uuid(value)
         if isinstance(value, str):

--- a/ulid/__init__.py
+++ b/ulid/__init__.py
@@ -2,6 +2,7 @@ from __future__ import annotations
 
 import functools
 import os
+import sys
 import time
 import uuid
 from datetime import datetime
@@ -13,7 +14,11 @@ from typing import Generic
 from typing import TYPE_CHECKING
 from typing import TypeVar
 
-from typing_extensions import Self
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
 
 from ulid import base32
 from ulid import constants

